public/app/plugins/panel/graph3/module.tsx

```diff
@@ -0,0 +1,177 @@
+import { PanelPlugin } from '@grafana/data';
+import { GraphPanel } from './GraphPanel';
+import { Options } from './types';
+import { GraphCustomFieldConfig } from '@grafana/ui';
+
+export const plugin = new PanelPlugin<Options, GraphCustomFieldConfig>(GraphPanel)
+  .useFieldConfig({
+    useCustomConfig: builder => {
+      builder
+        .addBooleanSwitch({
+          path: 'line.show',
+          name: 'Show lines',
+          description: '',
+          defaultValue: true,
+        })
+        .addSelect({
+          path: 'line.width',
+          name: 'Line width',
```

Comment:
do we need the nesting of properties here? Flat properties are usually easier to work with

Reply:
yes, this is what I started with, but since there are plenty of "related properties I find it easier to have specific types in the interface